src/plugin/fieldViews/ProseMirrorFieldView.ts

```diff
@@ -145,62 +147,99 @@ export abstract class ProseMirrorFieldView extends FieldView<string> {
 
-    if (diffStart === null) {
-      return this.maybeRerenderDecorations();
-    }
+    if (diffStart && diffEnd) {
+      let { a: endOfOuterDiff, b: endOfInnerDiff } = diffEnd;
+      // This overlap accounts for a situation where we're diffing nodes where we encounter
+      // identical content.
+      //
+      // For example, if the inner node has content 'a', and the outer node has content 'aa',
+      // `diffStart` sees (numbers are positions, ^ denotes the value found by the method)
+      //
+      // ab    inner node
+      // abb   outer node
+      // 123
+      //   ^
+      //
+      // `diffEnd` for the inner node is
+      //  ab   inner node
+      // abb   outer node
+      // 123
+      //  ^
+      //
+      // But 2 is not the correct end of the diff. The correct diff is (3, 3).
+      //
+      // This happens because `findDiffEnd` finds the first point at which the content differs,
+      // starting from the end of the nodes. So if we encounter identical content, the diff will
+      // be shorter by the length of the identical content we encounter – or the overlap between
+      // the two nodes from the point of view of the diff.
+      const overlap = diffStart - Math.min(endOfOuterDiff, endOfInnerDiff);
+      if (overlap > 0) {
+        endOfOuterDiff += overlap;
+        endOfInnerDiff += overlap;
+      }
 
-    const diffEnd = node.content.findDiffEnd(state.doc.content);
-    if (!diffEnd) {
-      // There's no difference between these nodes.
-      return this.maybeRerenderDecorations();
+      shouldDispatchTransaction = true;
+      tr = tr.replace(
+        diffStart,
+        endOfInnerDiff,
+        node.slice(diffStart, endOfOuterDiff)
+      );
     }
-
-    let { a: endOfOuterDiff, b: endOfInnerDiff } = diffEnd;
-
-    // This overlap accounts for a situation where we're diffing nodes where we encounter
-    // identical content.
-    //
-    // For example, if the inner node has content 'a', and the outer node has content 'aa',
-    // `diffStart` sees (numbers are positions, ^ denotes the value found by the method)
-    //
-    // ab    inner node
-    // abb   outer node
-    // 123
-    //   ^
-    //
-    // `diffEnd` for the inner node is
-    //  ab   inner node
-    // abb   outer node
-    // 123
-    //  ^
-    //
-    // But 2 is not the correct end of the diff. The correct diff is (3, 3).
-    //
-    // This happens because `findDiffEnd` finds the first point at which the content differs,
-    // starting from the end of the nodes. So if we encounter identical content, the diff will
-    // be shorter by the length of the identical content we encounter – or the overlap between
-    // the two nodes from the point of view of the diff.
-    const overlap = diffStart - Math.min(endOfOuterDiff, endOfInnerDiff);
-    if (overlap > 0) {
-      endOfOuterDiff += overlap;
-      endOfInnerDiff += overlap;
+    if (shouldDispatchTransaction) {
```

Comment:
More a note to self: shame we can't ask the transaction itself if it's changed! (There's a `this.updated` field, but it's private.) We might be able to avoid this state with ('transaction selection has changed' && 'transaction.steps.length > 0), but it's a bit of a weird one as it relies on the dev knowing that `.setSelection` mutates the selection, and `.replace` adds steps. So this is fine.